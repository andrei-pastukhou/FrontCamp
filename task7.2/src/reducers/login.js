const initialState = {
    isLogin: false,
    token: '',
    message: '',
    pending: false,
    username: '',
}
const login = (state = initialState, action) => {
    switch (action.type) {
        case 'LOGIN':
            return (
                Object.assign( {}, state,
                    {
                        isLogin: false,
                        pending: true,
                    }
                )
            );
        case 'LOGIN_SUCCSESS':
            return (
                Object.assign({}, state,
                    {
                        isLogin: true,
                        pending: false,
                        token: action.token,
                        message: '',
                        username: action.username

                    }
                )
            );
        case 'LOGIN_ERROR':
            return (
                Object.assign({}, state,
                    {
                        isLogin: false,
                        pending: false,
                        token: '',
                        message: action.message,
                        username: ''
                    }
                )
            );
        case 'LOGOUT':
            return (
                Object.assign({}, state,
                    {
                        isLogin: false,
                        pending: false,
                        token: '',
                        message: '',
                        username: ''
                    }
                )
            );
        default:
            return state
    }
};

export default login
