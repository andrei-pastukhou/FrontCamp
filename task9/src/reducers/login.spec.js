import login from './login'

describe('login reducer', () => {
    it('should handle initial state', () => {
        expect(
        login(undefined, {})
        ).toEqual(
            {
                isLogin: false,
                token: '',
                message: '',
                pending: false,
                username: '',
            }
        )
    });
    it('should handle LOGIN', () => {
        expect(
        login([], {
            type: 'LOGIN',
        })
        ).toEqual(
            {
                isLogin: false,
                pending: true,
            }
        )
    });
    it('should handle LOGIN_SUCCSESS', () => {
        expect(
        login([], {
            type: 'LOGIN_SUCCSESS',
            token: 'test_token',
            username: 'test_username'
        })
        ).toEqual(
            {
                isLogin: true,
                pending: false,
                token: 'test_token',
                message: '',
                username: 'test_username'
            }
        )
    });
    it('should handle LOGIN_ERROR', () => {
        expect(
        login([], {
            type: 'LOGIN_ERROR',
            message: 'test error login',

        })
        ).toEqual(
            {
                isLogin: false,
                pending: false,
                token: '',
                message: 'test error login',
                username: ''
            }
        )
    });
    it('should handle LOGOUT', () => {
        expect(
        login([], {
            type: 'LOGOUT',
            message: 'test error login',

        })
        ).toEqual(
            {
                isLogin: false,
                pending: false,
                token: '',
                message: '',
                username: ''
            }
        )
    });
});

describe('login reduxer with snapshots ', () => {
    it('+++ reducer with shapshot', () => {
      expect(login(undefined, { type: 'default' })).toMatchSnapshot();
    });
  
    it('+++ reducer with shapshot', () => {
      const action = {
        type: 'LOGIN',
      };
      expect(login(undefined, action)).toMatchSnapshot();
    });
   
    it('should handle LOGIN_SUCCSESS', () => {
        const action = {
            type: 'LOGIN_SUCCSESS',
            token: 'test_token',
            username: 'test_username'
        }
        expect(login(undefined, action)).toMatchSnapshot();
    });
    it('should handle LOGIN_ERROR', () => {
        const action = {
            type: 'LOGIN_ERROR',
            message: 'test error login',
        }
        expect(login(undefined, action)).toMatchSnapshot();
    });
    it('should handle LOGOUT', () => {
        const action =  {
            type: 'LOGOUT',
            message: 'test error login',
        }
        expect(login(undefined, action)).toMatchSnapshot();
    });
  });