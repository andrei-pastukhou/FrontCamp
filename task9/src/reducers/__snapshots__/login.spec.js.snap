// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login reduxer with snapshots  +++ reducer with shapshot 1`] = `
Object {
  "isLogin": false,
  "message": "",
  "pending": false,
  "token": "",
  "username": "",
}
`;

exports[`login reduxer with snapshots  +++ reducer with shapshot 2`] = `
Object {
  "isLogin": false,
  "message": "",
  "pending": true,
  "token": "",
  "username": "",
}
`;

exports[`login reduxer with snapshots  should handle LOGIN_ERROR 1`] = `
Object {
  "isLogin": false,
  "message": "test error login",
  "pending": false,
  "token": "",
  "username": "",
}
`;

exports[`login reduxer with snapshots  should handle LOGIN_SUCCSESS 1`] = `
Object {
  "isLogin": true,
  "message": "",
  "pending": false,
  "token": "test_token",
  "username": "test_username",
}
`;

exports[`login reduxer with snapshots  should handle LOGOUT 1`] = `
Object {
  "isLogin": false,
  "message": "",
  "pending": false,
  "token": "",
  "username": "",
}
`;
